features/admin.workflow-models.v1 /components/create/configuration-details-form.tsx

```diff
@@ -0,0 +1,153 @@
+import { PlusIcon } from "@oxygen-ui/react-icons";
+import React, {
+    ForwardedRef,
+    forwardRef,
+    ForwardRefExoticComponent,
+    ReactElement,
+    RefAttributes,
+    useCallback,
+    useEffect,
+    useImperativeHandle,
+    useRef,
+    useState
+} from "react";
+import { RolesInterface } from "@wso2is/core/models";
+import Button from "@oxygen-ui/react/Button";
+import "./configuration-details-form.scss";
+import Box from "@oxygen-ui/react/Box";
+import { UserBasicInterface } from "@wso2is/admin.core.v1/models/users";
+import ApprovalStep from "./approval-step";
+import { ConfigurationsFormValuesInterface } from "../../models/ui";
+
+export interface ConfigurationsPropsInterface {
+    isReadOnly?: boolean;
+    onSubmit?: (values: ConfigurationsFormValuesInterface) => void;
+    initialValues?: Partial<ConfigurationsFormValuesInterface>;
+}
+
+export interface ConfigurationsFormRef {
+    triggerSubmit: () => void;
+}
+
+export type MultiStepApprovalTemplate = {
+    id: string;
+    stepNumber: number;
+    roles: string[];
+    users: string[];
+};
+
+const ConfigurationsForm: ForwardRefExoticComponent<RefAttributes<ConfigurationsFormRef> &
+    ConfigurationsPropsInterface> = forwardRef(
+    ({ isReadOnly, onSubmit, initialValues }: ConfigurationsPropsInterface, ref: ForwardedRef<ConfigurationsFormRef>): ReactElement => {
+        const triggerFormSubmit = useRef<() => void>();
+
+        useEffect(() => {
+            if (initialValues?.approvalSteps?.length) {
+                const parsedSteps = initialValues.approvalSteps.map((step, index) => ({
+                    id: `step${Date.now()}-${index}`,
```

Comment:
```suggestion
                    id: `step-${index}`,
```
Do we need the `Date` here? can we only use index as the step?

Reply:
If Date.now() is removed and only index used, there may be issues with duplicate IDs in dynamic lists, especially when adding or deleting steps.